app/src/main/java/com/iti/android/team1/ebuy/model/datasource/repository/Repository.kt

```diff
@@ -2,6 +2,15 @@ package com.iti.android.team1.ebuy.model.datasource.repository
 
 import com.iti.android.team1.ebuy.model.datasource.remotesource.RemoteSource
 import com.iti.android.team1.ebuy.model.datasource.remotesource.RetrofitHelper
+import com.iti.android.team1.ebuy.model.pojo.Products
 
 class Repository(private val remoteSource:RemoteSource = RetrofitHelper) : IRepository {
+    override suspend fun getBrandProducts(brandID: Long): Products? {
+        val response = remoteSource.getBrandProducts(brandID)
+        if(response.isSuccessful){
+            return response.body()
+        }else{
+            throw Exception("${response.errorBody()}")
+        }
+    }
```

Comment:
we should return Error so view model can handle it so we must create sealed class and return it
you can see it in my branch 
what is your opinion?